src/logics/graphImpl/sigmaImpl.ts

```diff
@@ -41,28 +41,52 @@ function createSigmaGraph(container: HTMLElement) {
   let draggedNode: string | null = null;
   let isDragging = false;
 
+  //State for creating edge
+  let shiftKeyDown = false;
+  let draggingEdge = false;
+  let startNode : string | null = null;
+
   // On mouse down on a node
   //  - we enable the drag mode
   //  - save in the dragged node in the state
   //  - highlight the node
   //  - disable the camera so its state is not updated
   sigma.on("downNode", (e) => {
-    sigmaLayout?.stop()
-    store.dispatch(setIsPhysicsEnabled(false))
-    isDragging = true;
-    draggedNode = e.node;
-    graph!.setNodeAttribute(draggedNode, "highlighted", true);
+    if (shiftKeyDown) {
+      sigmaLayout?.stop()
+      store.dispatch(setIsPhysicsEnabled(false))
+      draggingEdge = true;
+      startNode = e.node;
+    }
+    else {
+      sigmaLayout?.stop()
+      store.dispatch(setIsPhysicsEnabled(false))
```

Comment:
Since
```
      sigmaLayout?.stop()
      store.dispatch(setIsPhysicsEnabled(false))
```
is being called in both branches of the if-else. You can just move these two lines outside of the if-else and cut down on code duplication.